### PR TITLE
You shouldn't need `--only` to only run `test.only` tests

### PR DESCRIPTION
**Pull request by Jarred-Sumner**
### What does this PR do?

Fixes https://github.com/oven-sh/bun/issues/10850
Fixes https://github.com/oven-sh/bun/issues/7701
Fixes https://github.com/oven-sh/bun/issues/9495

Currently, for the below test to pass, you need to start bun with `--only`:
```sh
bun test --only
```

This is a bad product decision. You shouldn't have to pass `--only`. It should just detect you called `.only` and skip the tests that didn't.

```ts
test("this one should not run #1", () => {
  expect.unreachable();
});

test.only("only this one should run", () => {
  expect().pass();
});

test("this one should not run #2", () => {
  expect.unreachable();
});
```

Originally, it was because we didn't enqueue each test and instead ran them immediately, but it was impossible to be compatible with Jest and work that way so we stopped doing that a long time ago (before the v0.1 public release)

This also resets the state of `test.only` at the beginning of each module. That means having one `test.only` in a later file no longer causes all subsequent files to be `.only`. To make all files run with `.only`, you can prefix `--only`. Since the filesystem sorting order was never deterministic, this shouldn't be a breaking change.

### How did you verify your code works?

Tests

### CODE REVIEW

**Comment by robobun** [comment]
@Jarred-Sumner, your commit 5e7cf676a28c3c6588b92aa81f9700a51845d0c8 is building: [`#2194`](https://buildkite.com/bun/bun/builds/2194)

File: test/js/bun/test/only-fixture-1.ts [hunk 1]
```diff
@@ -0,0 +1,11 @@
+test.only("only this one should run", () => {
+  expect().pass();
```
Comment (Electroid) [inline test/js/bun/test/only-fixture-1.ts:2]: This doesn't actually verify the code ran, it would be better to `console.log` and check the output.

**Review by dylan-conway — CHANGES_REQUESTED** [review]
The new tests are failing on linux and macos

File: test/js/bun/test/only-fixture-1.ts (hunk 1 above)
Comment (Jarred-Sumner) [reply test/js/bun/test/only-fixture-1.ts:2]: We test the number of tests ran by the output

**Comment by Jarred-Sumner** [comment]
merging per IRL discussion

**Pull request merged**